File: 202101240037 - Kielmoritz (VbNet - Cargo Automation)/01_source-code/05_project/VBOtomasyon/VBOtomasyon/Class_Islemler.vb
Imports System.Collections.Generic
Imports System.Data.SqlClient
Imports System.Data
Imports System.Collections
Imports System
Imports System.Windows.Forms
Imports System.Security.Cryptography
Imports System.Text


Class Class_Islemler
    Private baglanti As SqlConnection = New SqlConnection("Data Source =localhost, 1433; Network Library = DBMSSOCN; Initial Catalog =G2FinalKargo; User ID=remote; Password=celeron;")

    Public Function BaglantiTest() As Boolean
        Try
            baglanti.Open()
            baglanti.Close()
            Return True
        Catch
            baglanti.Close()
            Return False
        End Try
    End Function


    Public Function Md5Uret(theInput As String) As String
        Using hasher As MD5 = MD5.Create()
            Dim dbytes As Byte() =
             hasher.ComputeHash(Encoding.UTF8.GetBytes(theInput))
            Dim sBuilder As New StringBuilder()
            For n As Integer = 0 To dbytes.Length - 1
                sBuilder.Append(dbytes(n).ToString("X2"))
            Next n
            Return sBuilder.ToString()
        End Using

    End Function

    Public Function Kullanici_Giris(ByVal KullaniciAd As String, ByVal Sifre As String) As Boolean
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand("Select User_Name From Login_Tablosu_F Where User_Name='" + KullaniciAd + "' and Password='" + Sifre + "'", baglanti)
            Dim oku As SqlDataReader = komut.ExecuteReader()

            If oku.Read() Then
                baglanti.Close()
                Return True
            End If

            baglanti.Close()
            Return False
        Catch
            baglanti.Close()
            Return False
        End Try
    End Function

    Public Function Kullanici_Kontrol(ByVal KullaniciAd As String) As Boolean
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand("Select User_Name From Login_Tablosu_F Where User_Name='" + KullaniciAd + "'", baglanti)
            Dim oku As SqlDataReader = komut.ExecuteReader()

            If oku.Read() Then
                baglanti.Close()
                Return True
            End If

            baglanti.Close()
            Return False
        Catch
            baglanti.Close()
            Return False
        End Try
    End Function
    Public Function Kullanici_SifreYenile(ByVal KullaniciAd As String, ByVal Sifre As String) As Boolean
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand("Update Login_Tablosu_F Set Password='" + Sifre + "' Where User_Name='" + KullaniciAd + "'", baglanti)
            komut.ExecuteNonQuery()

            baglanti.Close()
            Return True
        Catch
            baglanti.Close()
            Return False
        End Try
    End Function
    Public Function Kullanici_Getir(ByVal KullaniciAd As String) As String()
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand("Select Question,Answer From Login_Tablosu_F Where User_Name='" + KullaniciAd + "'", baglanti)
            Dim oku As SqlDataReader = komut.ExecuteReader()

            If oku.Read() Then
                Dim veriler As String() = New String() {oku("Question"), oku("Answer")}

                baglanti.Close()
                Return veriler
            End If

            baglanti.Close()
            Return Nothing
        Catch
            baglanti.Close()
            Return Nothing
        End Try
    End Function

    Public Function Kontrol(ByVal tablo As String, ByVal Id As Integer) As Boolean
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand(tablo + "_Getir", baglanti)
            komut.CommandType = CommandType.StoredProcedure
            komut.Parameters.AddWithValue("@Tablo", tablo)
            komut.Parameters.AddWithValue("@Id", Id)
            Dim oku As SqlDataReader = komut.ExecuteReader()

            If oku.Read() Then
                baglanti.Close()
                Return True
            End If

            baglanti.Close()
            Return False
        Catch
            baglanti.Close()
            Return False
        End Try
    End Function

    Public Function Ekle(ByVal tablo As String, ByVal veriler As ArrayList) As Boolean
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand(tablo & "_Ekle", baglanti)
            komut.CommandType = CommandType.StoredProcedure

            For Each Kolon As ArrayList In veriler
                komut.Parameters.AddWithValue("@" & Kolon(0), Kolon(1))
            Next

            komut.ExecuteNonQuery()
            komut.Dispose()
            baglanti.Close()
            Return True
        Catch
            baglanti.Close()
            Return False
        End Try
    End Function

    Public Function Guncelle(ByVal tablo As String, ByVal veriler As ArrayList) As Boolean
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand(tablo & "_Guncelle", baglanti)
            komut.CommandType = CommandType.StoredProcedure

            For Each Kolon As ArrayList In veriler
                komut.Parameters.AddWithValue("@" & Kolon(0), Kolon(1))
            Next

            komut.ExecuteNonQuery()
            komut.Dispose()
            baglanti.Close()
            Return True
        Catch
            baglanti.Close()
            Return False
        End Try
    End Function

    Public Function Sil(ByVal tablo As String, ByVal Id As Integer) As Boolean


        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand(tablo + "_Sil", baglanti)
            komut.CommandType = CommandType.StoredProcedure
            komut.Parameters.AddWithValue("@Tablo", tablo)
            komut.Parameters.AddWithValue("@Id", Id)
            komut.ExecuteNonQuery()
            komut.Dispose()
            baglanti.Close()
            Return True
        Catch
            baglanti.Close()
            Return False
        End Try
    End Function

    Public Function Getir(ByVal tablo As String, ByVal Id As Integer) As ArrayList
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand(tablo + "_Getir", baglanti)
            komut.CommandType = CommandType.StoredProcedure
            komut.Parameters.AddWithValue("@Tablo", tablo)
            komut.Parameters.AddWithValue("@Id", Id)
            Dim oku As SqlDataReader = komut.ExecuteReader()
            Dim veriler As ArrayList = New ArrayList()

            If oku.Read() Then
                Dim i As Integer = 0
                While True
                    Try
                        veriler.Add(oku(i))
                    Catch
                        Exit While
                        Exit Try
                    End Try
                    i += 1
                End While
            End If

            komut.Dispose()
            baglanti.Close()
            Return veriler
        Catch
            baglanti.Close()
            Return Nothing
        End Try
    End Function

    Public Function Getir(ByVal tablo As String, ByVal sorulan As String) As String()
        Try
            Dim veriler As DataSet = Kayitlar(tablo)
            Dim liste As ArrayList = Donustur(veriler)

            For Each items As String() In liste

                For Each veri As String In items
                    If veri = sorulan Then Return items
                Next
            Next

            Return Nothing
        Catch
            Return Nothing
        End Try
    End Function

    Public Function Kayitlar(ByVal tablo As String) As DataSet
        Try
            baglanti.Open()
            Dim komut As SqlCommand = New SqlCommand("Kayitlar", baglanti)
            komut.CommandType = CommandType.StoredProcedure
            komut.Parameters.AddWithValue("@Tablo", tablo)
            Dim da As SqlDataAdapter = New SqlDataAdapter(komut)
            Dim ds As DataSet = New DataSet()
            da.Fill(ds)
            komut.Dispose()
            baglanti.Close()
            Return ds
        Catch

            baglanti.Close()
            Return Nothing
        End Try
    End Function

    Public Function GonderiAra(ByVal aranan As String) As DataSet
        Try
            baglanti.Open()
            Dim sorgu As String = "Select * From Gonderi_Tablosu_F Where Gonderi_Tarihi LIKE '%" + aranan + "%' or Sube_No=" + aranan + " or Personel_No=" + aranan + " or Musteri_No=" + aranan + ""
            Dim komut As SqlCommand = New SqlCommand(sorgu, baglanti)
            Dim da As SqlDataAdapter = New SqlDataAdapter(komut)
            Dim ds As DataSet = New DataSet()
            da.Fill(ds)
            komut.Dispose()
            baglanti.Close()
            Return ds
        Catch

            baglanti.Close()
            Return Nothing
        End Try
    End Function

    Public Function Donustur(ByVal veri As DataSet) As ArrayList
        Try
            Dim veriler As ArrayList = New ArrayList()

            For Each satir As DataRow In veri.Tables(0).Rows
                Dim parcalanmis As String() = String.Join(",", satir.ItemArray).Split(",")
                veriler.Add(parcalanmis)
            Next
            Return veriler
        Catch
            Return Nothing
        End Try
    End Function

    Public Sub CBDoldur(ByVal box As ComboBox, ByVal tablo As String, ByVal kolon As Integer)
        Dim veriler As DataSet = Kayitlar(tablo)

        For Each satir As DataRow In veriler.Tables(0).Rows
            Dim veri As Object() = satir.ItemArray
            box.Items.Add(veri(kolon))
        Next
        box.SelectedIndex = 0
    End Sub

    Public Sub CBDoldur(ByVal box As ComboBox, ByVal tablo As String, ByVal kolonlar As Integer())
        Try
            Dim veriler As DataSet = Kayitlar(tablo)
            For Each satir As DataRow In veriler.Tables(0).Rows
                Dim veri As Object() = satir.ItemArray
                Dim eklenecek As String = ""
                For Each no In kolonlar
                    eklenecek += veri(no) + " "
                Next
                box.Items.Add(eklenecek)
            Next
            box.SelectedIndex = 0
        Catch ex As Exception
            MesajKutu(2, "listeleme")
        End Try
    End Sub

    Public Function Donustur(ByVal veri As DataSet, ByVal kolon As Integer) As List(Of String)
        Try
            Dim veriler As List(Of String) = New List(Of String)()

            For Each satir As DataRow In veri.Tables(0).Rows
                Dim satirVeriler As Object() = satir.ItemArray
                veriler.Add(satirVeriler(kolon))
            Next

            Return veriler
        Catch
            Return Nothing
        End Try
    End Function

    Public Sub Ekle(ByVal form As Form, ByVal tablo As String, ByVal veriler As ArrayList)
        If Ekle(tablo, veriler) Then
            MesajKutu(tablo & " ekleme")
        Else
            MesajKutu(2, tablo & " ekleme")
        End If

    End Sub

    Public Sub Guncelle(ByVal form As Form, ByVal tablo As String, ByVal veriler As ArrayList, ByVal Id As Integer)
        veriler.Add(New ArrayList() From {
            "Id",
            Id
        })

        If Guncelle(tablo, veriler) Then
            MesajKutu(tablo & " güncelleme")
        Else
            MesajKutu(2, tablo & " güncelleme")
        End If

        Return
    End Sub

    Public Sub Sil(ByVal form As Form, ByVal tablo As String, ByVal Id As Integer)
        If SoruKutu(tablo & " tablosundan " + Id.ToString + " nolu kaydı silmek") Then

            If Sil(tablo, Id) Then
                MesajKutu("silme")
                form.Close()
            Else
                MesajKutu(2, "silme")
            End If
        End If
    End Sub

    Public Sub CBSec(ByVal box As ComboBox, ByVal veriler As List(Of String), ByVal Id As Integer)
        For i As Integer = 0 To veriler.Count - 1
            If Id = Convert.ToInt32(veriler(i)) Then
                box.SelectedIndex = i
            End If
        Next
    End Sub

    Public Sub MesajKutu(ByVal tur As Integer, ByVal islem As String)
        Select Case tur
            Case 1
                MessageBox.Show(islem.ToUpper() & " işlemi için gerekli alanları doldurunuz.", "Uyarı", MessageBoxButtons.OK, MessageBoxIcon.Warning)
            Case 2
                MessageBox.Show(islem.ToUpper() & " işlemi başarısız !!!", "Hata", MessageBoxButtons.OK, MessageBoxIcon.[Error])
        End Select
    End Sub

    Public Sub MesajKutu(ByVal islem As String)
        MessageBox.Show(islem.ToUpper() & " işlemi başarılı.", "Bilgi", MessageBoxButtons.OK, MessageBoxIcon.Information)
    End Sub

    Public Function SoruKutu(ByVal soru As String) As Boolean
        Dim cevap As DialogResult = MessageBox.Show(soru.ToUpper() & " istediğinize emin misiniz?", "Soru", MessageBoxButtons.YesNo, MessageBoxIcon.Question)

        If cevap = DialogResult.Yes Then
            Return True
        Else
            Return False
        End If
    End Function


End Class

File: 202101112203 - Ecces-2 (MsSQL - Database Design)/01_source-code/05_project/GTS-Project/GTS-Project/Class_Islemler.cs
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Data;
using System.Collections;
using System;
using System.Windows.Forms;
using System.Linq;

namespace GTS_Project
{
    class Class_Islemler
    {
        //SqlConnection baglanti = new SqlConnection("Data Source=" + Ayarlar.Default.Sunucu + ";Initial Catalog=" + Ayarlar.Default.Veritabani + ";User Id="+Ayarlar.Default.Kullanici+";Password="+Ayarlar.Default.Parola+";");
        SqlConnection baglanti = new SqlConnection("Data Source = " + Ayarlar.Default.Sunucu + ", 1433; Network Library = DBMSSOCN; Initial Catalog = " + Ayarlar.Default.Veritabani + "; User ID = " + Ayarlar.Default.Kullanici + "; Password = " + Ayarlar.Default.Parola + ";");
        //BAĞLANTI TEST
        public bool BaglantiTest() // Veritabanına bağlantının gerçekleşme durumunu test etme
        {
            try
            {
                baglanti.Open();
                baglanti.Close();
                return true;
            }
            catch
            {
                baglanti.Close();
                return false;
            }
        }
        public bool Yonetici_Kontrol(string KullaniciAd, string Parola) //Yonetici bilgilerinin kontrolü
        {
            try
            {
                baglanti.Open();
                SqlCommand komut = new SqlCommand("Yonetici_Kontrol", baglanti);
                komut.CommandType = CommandType.StoredProcedure;
                komut.Parameters.AddWithValue("@KullaniciAd", KullaniciAd);
                komut.Parameters.AddWithValue("@Parola", Parola);
                SqlDataReader oku = komut.ExecuteReader();
                if (oku.Read())
                {
                    baglanti.Close();
                    return true;
                }
                baglanti.Close();
                return false;
            }
            catch
            {
                baglanti.Close();
                return false;
            }
        }
        //ANA İŞLEMLER
        public bool Kontrol(string tablo, int Id) //Tabloda istenilen kaydın olup olmadığı kontrolü
        {
            try
            {
                baglanti.Open();
                SqlCommand komut = new SqlCommand("Getir", baglanti);
                komut.CommandType = CommandType.StoredProcedure;
                komut.Parameters.AddWithValue("@Tablo", tablo);
                komut.Parameters.AddWithValue("@Id", Id);
                SqlDataReader oku = komut.ExecuteReader();
                if (oku.Read())
                {
                    baglanti.Close();
                    return true;
                }
                baglanti.Close();
                return false;
            }
            catch
            {
                baglanti.Close();
                return false;
            }
        }
        public bool Ekle(string tablo, ArrayList veriler) //Herhangibir tabloya yeni kayıt ekleme
        {
            try
            {

                baglanti.Open();
                SqlCommand komut = new SqlCommand(tablo + "_Ekle", baglanti);
                komut.CommandType = CommandType.StoredProcedure;
                foreach (ArrayList Kolon in veriler)
                    komut.Parameters.AddWithValue("@" + Kolon[0], Kolon[1]);
                komut.ExecuteNonQuery();
                komut.Dispose();
                baglanti.Close();
                return true;
            }
            catch
            {
                baglanti.Close();
                return false;
            }
        }
        public bool Guncelle(string tablo, ArrayList veriler) //Herhangibir tablodaki bir kaydı güncelleme
        {
            try
            {
                baglanti.Open();
                SqlCommand komut = new SqlCommand(tablo + "_Guncelle", baglanti);
                komut.CommandType = CommandType.StoredProcedure;
                foreach (ArrayList Kolon in veriler)
                    komut.Parameters.AddWithValue("@" + Kolon[0], Kolon[1]);
                komut.ExecuteNonQuery();
                komut.Dispose();
                baglanti.Close();
                return true;
            }
            catch
            {
                baglanti.Close();
                return false;
            }
        }
        public bool Sil(string tablo, int Id) //Herhangibir tablodaki 1 kaydı silme
        {
            if (!(Kontrol(tablo, Id)))
                return false;
            try
            {
                baglanti.Open();
                SqlCommand komut = new SqlCommand("Sil", baglanti);
                komut.CommandType = CommandType.StoredProcedure;
                komut.Parameters.AddWithValue("@Tablo", tablo);
                komut.Parameters.AddWithValue("@Id", Id);
                komut.ExecuteNonQuery();
                komut.Dispose();
                baglanti.Close();
                return true;
            }
            catch
            {
                baglanti.Close();
                return false;
            }
        }
        public ArrayList Getir(string tablo, int Id) //Herhangibir tablodan id bilgisine göre kayıt getirme
        {
            try
            {
                baglanti.Open();
                SqlCommand komut = new SqlCommand("Getir", baglanti);
                komut.CommandType = CommandType.StoredProcedure;
                komut.Parameters.AddWithValue("@Tablo", tablo);
                komut.Parameters.AddWithValue("@Id", Id);
                SqlDataReader oku = komut.ExecuteReader();
                ArrayList veriler = new ArrayList();
                if (oku.Read())
                {
                    for (int i = 0; true; i++)
                    {
                        try { veriler.Add(oku[i]); }
                        catch { break; }
                    }
                }

                komut.Dispose();
                baglanti.Close();
                return veriler;
            }
            catch
            {
                baglanti.Close();
                return null;
            }

        }
        public string[] Getir(string tablo, string sorulan) //Herhangibir tablodan herhangibir kolon verisine göre kayıt getirme
        {
            try
            {
                DataSet veriler = Kayitlar(tablo);
                ArrayList liste = Donustur(veriler);
                foreach (string[] items in liste)
                    foreach (string veri in items)
                        if (veri == sorulan)
                            return items;
                return null;
            }
            catch { return null; }
        }
        public DataSet Kayitlar(string tablo) //Herhangibir tablodaki tüm kayıtları listeleme
        {
            try
            {
                baglanti.Open();
                SqlCommand komut = new SqlCommand("Kayitlar", baglanti);
                komut.CommandType = CommandType.StoredProcedure;
                komut.Parameters.AddWithValue("@Tablo", tablo);
                SqlDataAdapter da = new SqlDataAdapter(komut);
                DataSet ds = new DataSet();
                da.Fill(ds);
                komut.Dispose();
                baglanti.Close();
                return ds;
            }
            catch
            {
                baglanti.Close();
                return null;
            }
        }
        public ArrayList Donustur(DataSet veri) //Dataset türündeki bir veriyi ArrayList türüne dönüştürme
        {
            try
            {
                ArrayList veriler = new ArrayList();
                foreach (DataRow satir in veri.Tables[0].Rows)
                {
                    string[] parcalanmis = string.Join(",", satir.ItemArray).Split(',').ToArray();
                    veriler.Add(parcalanmis);
                }
                return veriler;
            }
            catch { return null; }
        }
        public List<string> Donustur(DataSet veri, int kolon) //Dataset türündeki bir veriden tek bir kolonu List<string> türüne dönüştürme
        {
            try
            {
                List<string> veriler = new List<string>();
                foreach (DataRow satir in veri.Tables[0].Rows)
                {
                    string[] parcalanmis = string.Join(",", satir.ItemArray).Split(',').ToArray();
                    veriler.Add(parcalanmis[kolon]);
                }
                return veriler;
            }
            catch { return null; }
        }
        public List<string> Donustur(DataSet veri, int kolon, int f_kolon, int f_Id) //Dataset türündeki bir veriden tek bir kolonu List<string> türüne dönüştürme
        {
            try
            {
                List<string> veriler = new List<string>();
                foreach (DataRow satir in veri.Tables[0].Rows)
                {
                    if (Convert.ToInt32(satir.ItemArray[f_kolon]) == f_Id)
                    {
                        string[] parcalanmis = string.Join(",", satir.ItemArray).Split(',').ToArray();
                        veriler.Add(parcalanmis[kolon]);
                    }
                }
                return veriler;
            }
            catch { return null; }
        }
        public void Ekle(Form form, string tablo, ArrayList veriler)//Çağrılan formda verilen bilgilere göre hata veya uyarı mesaj kutusu oluşturma
        {
            if (Ekle(tablo, veriler))
                MesajKutu(tablo + " ekleme");
            else
                MesajKutu(2, tablo + " ekleme");
            Temizle(form);
        }
        public void Guncelle(Form form, string tablo, ArrayList veriler, int Id)//Herhangi bir tablodaki bir kaydı güncelleme ve mesaj kutusu oluşturma
        {
            veriler.Add(new ArrayList() { "Id", Id });
            if (Guncelle(tablo, veriler))
                MesajKutu(tablo + " güncelleme");
            else
                MesajKutu(2, tablo + " güncelleme");
            return;
        }
        public void Sil(Form form, string tablo, int Id)//Herhangi bir tablodaki bir kaydı silme ve mesaj kutusu oluşturma
        {
            if (SoruKutu(tablo + "'i silmek"))
            {
                if (Sil(tablo, Id))
                {
                    MesajKutu("silme");
                    form.Close();
                }
                else
                    MesajKutu(2, "silme");
            }
        }
        public void MesajKutu(int tur, string islem)//Hata ve Uyarı türünde mesaj kutuları oluşturma
        {
            switch (tur)
            {
                case 1:
                    MessageBox.Show(islem.ToUpper() + " işlemi için gerekli alanları doldurunuz.", "Uyarı", MessageBoxButtons.OK, MessageBoxIcon.Warning);
                    break;
                case 2:
                    MessageBox.Show(islem.ToUpper() + " işlemi başarısız !!!", "Hata", MessageBoxButtons.OK, MessageBoxIcon.Error);
                    break;
            }
        }
        public void MesajKutu(string islem)//Bilgi türünde mesaj kutusu oluşturma
        {
            MessageBox.Show(islem.ToUpper() + " işlemi başarılı.", "Bilgi", MessageBoxButtons.OK, MessageBoxIcon.Information);
        }
        public bool SoruKutu(string soru)//Soru türünde mesaj kutusu oluşturma
        {
            DialogResult cevap = MessageBox.Show(soru.ToUpper() + " istediğinize emin misiniz?", "Soru", MessageBoxButtons.YesNo, MessageBoxIcon.Question);
            if (cevap == DialogResult.Yes)
                return true;
            else
                return false;
        }
        public void CBSec(ComboBox box, List<string> veriler, int Id)//Herhangi bir Combobox nesnesinin verisetinden index seçme
        {
            for (int i = 0; i < veriler.Count; i++)
                if (Id == Convert.ToInt32(veriler[i]))
                    box.SelectedIndex = i;
        }
        public void CBDoldur(ComboBox box, string tablo, int kolon)//Herhangi bir Combobox nesnesine bir tablonun bir kolonundan veri aktarma
        {
            box.Items.Clear();
            try
            {
                ArrayList veriler = Donustur(Kayitlar(tablo));
                foreach (string[] item in veriler)
                    box.Items.Add(item[kolon]);
                box.SelectedIndex = 0;
            }
            catch
            {
                MesajKutu(2, tablo + " kaydı listeleme");
            }
        }
        public void CBDoldur(ComboBox box, string tablo, int kolon, int k_kolon, int k_Id)//Herhangi bir Combobox nesnesine bir tablonun bir kolonundan veri aktarma seçmeli
        {
            try
            {
                box.Items.Clear();
                ArrayList veriler = Donustur(Kayitlar(tablo));
                foreach (string[] item in veriler)
                {
                    if (Convert.ToInt32(item[k_kolon]) == k_Id)
                        box.Items.Add(item[kolon]);
                }

                box.SelectedIndex = 0;
            }
            catch
            {
                MesajKutu(2, tablo + " kaydı listeleme");
            }
        }
        public void CBDoldur(ComboBox box, string tablo, int[] kolonlar)//Herhangi bir Combobox nesnesine bir tablonun birçok kolonundan veri aktarma
        {
            try
            {
                box.Items.Clear();
                ArrayList siniflar = Donustur(Kayitlar(tablo));
                foreach (string[] item in siniflar)
                {
                    string veri = "";
                    for (int i = 0; i < kolonlar.Length; i++)
                    {
                        veri += item[kolonlar[i]] + " ";
                    }
                    box.Items.Add(veri);
                }

                box.SelectedIndex = 0;
            }
            catch
            {
                MesajKutu(2, tablo + " kaydı listeleme");
            }
        }

        public void Temizle(Form form)//Herhangi bir formdaki doldurulabilir nesnelerin içeriğini temizleme
        {
            foreach (Control item in form.Controls)
            {
                if (item is TextBox)
                    ((TextBox)item).Clear();
                if (item is MaskedTextBox)
                    ((MaskedTextBox)item).Clear();
                if (item is ComboBox)
                    if (((ComboBox)item).Items.Count > 0)
                        ((ComboBox)item).SelectedIndex = 0;
                if (item is DateTimePicker)
                    ((DateTimePicker)item).ResetText();
            }
        }

    }
}
